New: Option to enable multirom kmsg logging + history

nkk71 opened the pull request:
Adds the following to multirom-klogs dir:
- 5 last_kmsg log history
- current klog upon entering multirom
- current klog upon exiting multirom
- current kexc log

needs to be enabled in multirom.ini settings:
- enable_kmsg_logging=1
  or
- if you want to do it in TWRP
  eg https://github.com/nkk71/android_bootable_recovery/commit/497ce002dea9e80a9377f0f2d78e962d457fa681

Tasssadar commented:
Thanks!

Multirom uses 4 spaces instead of tabs for indentation, could you please convert them in your PR, so that we don't end up looking like TWRP? ^^

nkk71 commented:
sure thing, will let you know when it's changed

nkk71 committed:
New: Option to enable multirom kmsg logging + history

Adds the following to multirom-klogs dir:
* 5 last_kmsg log history
* current klog upon entering multirom
* current klog upon exiting multirom
* current kexc log

needs to be enabled in multirom.ini settings:
* enable_kmsg_logging=1
or
* if you want to do it in TWRP
eg https://github.com/nkk71/android_bootable_recovery/commit/497ce002dea9e80a9377f0f2d78e962d457fa681

nkk71 commented:
@Tasssadar 
changed the indentation to be inline with the rest of the code

PS: feel free to move the define and enum to .h
the only reason i left/keep them in the main code / close to the actual function is for future cherry picks, in case it is never merged into the main

nkk71 committed:
Update multirom logging

nkk71 commented:
deprecated.
